packages/Main/src/Layer/OGC3DTilesLayer.js

```diff
@@ -335,20 +333,58 @@ class OGC3DTilesLayer extends GeometryLayer {
      */
     _setupCacheAndQueues(view) {
         const id = view.id;
+        // Share the caches and queues to cut down on memory and correctly prioritize downloads.
+        // https://github.com/NASA-AMMOS/3DTilesRendererJS?tab=readme-ov-file#multiple-tilesrenderers-with-shared-caches-and-queues
         if (viewers[id]) {
+
             this.tilesRenderer.lruCache = viewers[id].lruCache;
             this.tilesRenderer.downloadQueue = viewers[id].downloadQueue;
             this.tilesRenderer.parseQueue = viewers[id].parseQueue;
+
+            // Store references to all layer callbacks for this view
+            // Each tile has it's own callback
+            if (!viewers[id].layerCallbacks) {
+                viewers[id].layerCallbacks = [];
+            }
+            viewers[id].layerCallbacks.push(this._originalSchedulingCallback);
+
+            // Create a combined callback that calls all layer callbacks
+            const combinedCallback = (func) => {
+                if (viewers[id] && viewers[id].layerCallbacks) {
+                    // Filter out null/undefined callbacks (from deleted layers)
+                    viewers[id].layerCallbacks = viewers[id].layerCallbacks.filter(callback => callback != null);
+                    viewers[id].layerCallbacks.forEach((callback) => {
+                        if (callback) { callback(func); }
+                    });
+                }
+            };
+
+            // Set the combined callback
+            // We set our scheduling callback for tiles downloading and parsing -> MANDATORY for VR
+            // (WebXR session has its own requestAnimationFrame method separate from that of the window
+            //  https://github.com/NASA-AMMOS/3DTilesRendererJS/issues/213#issuecomment-947943386)
+            // Necessary to update rendering of the tiles in VR
+            // Example: https://github.com/NASA-AMMOS/3DTilesRendererJS/blob/de25d27dc0e75278962b5f401faee30f8dce2fe0/example/vr.js
+            this.tilesRenderer.downloadQueue.schedulingCallback = combinedCallback;
+            this.tilesRenderer.parseQueue.schedulingCallback = combinedCallback;
         } else {
+            // Set the combined callback (cf. above comment)
+            this.tilesRenderer.downloadQueue.schedulingCallback = this.tilesSchedulingCB;
+            this.tilesRenderer.parseQueue.schedulingCallback = this.tilesSchedulingCB;
```

Comment:
Would it be better to create a "combined" callback that we'll not reassign?
I imagine that such callback code will call all `viewers[id].layerCallbacks`:
```suggestion
            const combined = (func) => {
              viewers[id].layersCallbacks.forEarch((cb) => cb(func));
            }
            // Set the combined callback (cf. above comment)
            this.tilesRenderer.downloadQueue.schedulingCallback = combined;
            this.tilesRenderer.parseQueue.schedulingCallback = combined;
```

When `viewer[id]` is defined, we'll only have to add the "layer" callback to the list of `viewers[id].layerCallbacks`.

Reply:
Thanks for the review !

What is the difference with what it's done here :

```
   }
            viewers[id].layerCallbacks.push(this._originalSchedulingCallback);

            // Create a combined callback that calls all layer callbacks
            const combinedCallback = (func) => {
                if (viewers[id] && viewers[id].layerCallbacks) {
                    // Filter out null/undefined callbacks (from deleted layers)
                    viewers[id].layerCallbacks = viewers[id].layerCallbacks.filter(callback => callback != null);
                    viewers[id].layerCallbacks.forEach((callback) => {
                        if (callback) { callback(func); }
                    });
                }
            };
            ```
            Currently I add the cb in layerCallbacks but i also remove the ones that are null or undefined ( as a safety)

Reply:
We don't have to re-assign the callback each time (see line 368-369) we add a new layer, but only once when we add new layer from a new view. It would make the code more readable I think.